packages/utils/telemetry-utils/src/errorLogging.ts

```diff
@@ -265,6 +266,30 @@ export function isTaggedTelemetryPropertyValue(x: any): x is ITaggedTelemetryPro
     return (typeof (x?.value) !== "object" && typeof (x?.tag) === "string");
 }
 
+/**
+ * Filter loggable telemetry properties
+ * @param x - any telemetry prop
+ * @returns - as-is if x is primitive. returns stringified if x is an array of primitive.
+ * otherwise returns null since this is what we support at the moment.
+ */
+// eslint-disable-next-line @rushstack/no-new-null
+function filterValidTelemetryProps(x: any): TelemetryEventPropertyType | null {
+    switch (typeof x) {
+        case "string":
+        case "number":
+        case "boolean":
+        case "undefined":
+            return x;
+        default:
+            if (!Array.isArray(x)) {
+                return null;
+            }
+            if (x.every((val) => filterValidTelemetryProps(val) !== null)) {
```

Comment:
i worry about performance here. this is a pretty costly call that will cause multiple enumerations of the array, check, then serialize. why not just force the consumer to pre-serialize?

Reply:
hmm i'm not sure how to force the consumer to pre-serialize, 
also i can remove the recursion and only support array of primitives, not array of arrays

Reply:
It's on an error codepath so I wasn't being very critical for perf optimization.

As for why the consumer can't pre-serialize:  This is a very awkward situation API-wise:  `LoggingError` logs all properties that aren't explicitly omitted.  ODSP Utils tacks on `facetCodes` for runtime use later.  It's not of an approved "Telemetry Properties" type (it's a string array), because that's what's needed at runtime.  And there's no type safety for this kind of thing.  The old way of dealing with it was just to discard the thing.  So to put it bluntly, this code is trying to support an unsupported thing.

Some other approaches to consider (Si and I discussed most of these already):
- Give `LoggingError` a function `addNonTelemetryProperty` or something that also omits the property from logging
- Don't log the "redacted" phrase at all and just drop things that are like this (rather than supporting this)
- Add a new column "redactedProperties" and put a serialized array in there (or use this code to serialize it, har har)
- Add full support for simple arrays and objects (see AB#1104) as telemetry properties (or an augementation we use only for TelemetryLogger)
- Leave it as-is

Reply:
And yeah I would remove the recursion because I think it will allow `[ "hello", [ "nested", "world" ] ]`